Refactor AnnotationBody to handle SpecificResource functionality

JulieWinchester opened the pull request:
This PR:
* Refactors `AnnotationBody` to support SpecificResource functionality. An annotation body with type "SpecificResource" can be instantiated as `AnnotationBody` and source values are accessible through the AnnotationBody object. `Annotation.getBody` methods have been changed to always return `AnnotationBody` objects like in default manifesto. `SpecificResource` as a class has been retained for annotation target methods, and those methods have not been changed. 
* Refactors the use of object variable flags to specify object type and identity (e.g., `AnnotationBody.isModel` or `Light.isLight`). In some cases these flags are unnecessary, like whether an object is a light can be tested with native JS using `obj instanceof Light`, etc. In other cases where the flags are used for type checking (`isModel`, `isPerspectiveCamera`, etc.), it seems better to use the approach already used in default manifesto for `ManifestResource` and elsewhere, where type checking is handled through methods instead of variables, e.g., `isPerspectiveCamera()` instead of `isPerspectiveCamera`. Default manifesto seems to store few persistent variables on objects and instead manages almost everything through object methods, so this PR tries to match that approach.
* Updates tests, all tests are successfully passing.  

The PR has been tested against manifold and Universal Viewer, and with one small additional tweak (backporting a `getRights()` method from latest manifesto), it works well with both downstream packages.

JulieWinchester committed:
Fix failing test

JulieWinchester committed:
Refactor AnnotationBody to handle SpecificResource content

vincentmarchetti committed:
Downgraded the dependency on npm isomorphic-unfetch from ^4 to ^3 .
This was done to improve compatibility with the UV (Universal Viewer)
webpack-dev

The dependency of the main manifesto (Presentaation 3 code) on isomorphic-unfetch
is also set to v 3 at this time.

JulieWinchester committed:
Merge remote-tracking branch 'origin/downgrade-isomorphic-unfetch' into annotation_body

vincentmarchetti reviewed (commented):
acknowledged, have checked out for testing

JulieWinchester committed:
Add AnnotationBody getResourceID, getSource can handle source as URI, and getFormat pulls from source or self

JulieWinchester commented:
Added a new commit with the following changes:
* Improved AnnotationBody `getSource()` to be able to work with source values that are string URIs. The return value of this new method is now `AnnotationBody | string | null`.
* Added a new AnnotationBody method `getResourceID()`. This method gets the most likely URI for content in an annotation body, which is for a content resource the body ID and for a specific resource the source's ID or the source's URI value (if source value is string). This is necessary for downstream packages like manifold and Universal Viewer to easily find the content URI. 
* AnnotationBody `getFormat()` now pulls from source for specific resource as well as from content resource value.

vincentmarchetti commented:
Have verified that the x3d based viewer can be modified to use the revised AnnotationBody class, checking that no functionality was lost in the changes.
